video/out/hwdec/hwdec_vaapi.c

```diff
@@ -156,19 +169,38 @@ static int init(struct ra_hwdec *hw)
 {
     struct priv_owner *p = hw->priv;
 
-    if (!ra_is_gl(hw->ra) || !eglGetCurrentContext())
-        return -1;
+#if HAVE_GL
+    if (ra_is_gl(hw->ra)) {
+        if (!eglGetCurrentContext())
+            return -1;
 
-    const char *exts = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS);
-    if (!exts)
-        return -1;
+        const char *exts = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS);
+        if (!exts)
+            return -1;
 
-    GL *gl = ra_gl_get(hw->ra);
-    if (!strstr(exts, "EXT_image_dma_buf_import") ||
-        !strstr(exts, "EGL_KHR_image_base") ||
-        !strstr(gl->extensions, "GL_OES_EGL_image") ||
-        !(gl->mpgl_caps & MPGL_CAP_TEX_RG))
-        return -1;
+        GL *gl = ra_gl_get(hw->ra);
+        if (!strstr(exts, "EXT_image_dma_buf_import") ||
+            !strstr(exts, "EGL_KHR_image_base") ||
+            !strstr(gl->extensions, "GL_OES_EGL_image") ||
+            !(gl->mpgl_caps & MPGL_CAP_TEX_RG))
+            return -1;
+
+        MP_VERBOSE(hw, "using VAAPI EGL interop\n");
+    }
+#endif
+
+#if HAVE_VULKAN
+    if (ra_pl_get(hw->ra)) {
+        const struct pl_gpu *gpu = ra_pl_get(hw->ra);
+        if (!(gpu->import_caps.buf & PL_HANDLE_DMA_BUF)) {
```

Comment:
You import the dmabuf as a texture, not a buffer, so you should be checking `import_caps.tex`